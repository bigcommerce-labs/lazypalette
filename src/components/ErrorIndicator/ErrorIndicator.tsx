import React, { SFC } from 'react';

import ButtonInput from '../ButtonInput/ButtonInput';

import { Wrapper } from './styles';

interface ErrorIndicatorProps {
    errors: Error[];
    clearErrors(): void;
}

export const ErrorIndicator: SFC<ErrorIndicatorProps> = ({ clearErrors, errors }): JSX.Element => {
    return (
        <Wrapper>
            {errors.map(error => (
                <p key={error.toString()}>{error.toString()}</p>
            ))}
            <ButtonInput onClick={clearErrors}>
                OK
            </ButtonInput>
        </Wrapper>
    );
};
