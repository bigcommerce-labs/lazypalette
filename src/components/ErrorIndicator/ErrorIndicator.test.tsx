import { shallow } from 'enzyme';
import React from 'react';

import { ErrorIndicator } from './ErrorIndicator';

describe('ErrorIndicator', () => {
    let clearErrors: any;

    beforeEach(() => {
        clearErrors = jest.fn();
    });

    describe('when rendering', () => {
        it('renders with no errors', () => {
            const errorIndicator = shallow(
                <ErrorIndicator clearErrors={clearErrors} errors={[]}/>);
            expect(errorIndicator).toMatchSnapshot();
        });

        it('renders with errors', () => {
            const errors = [
                new Error('hello'),
                new Error('goodbye'),
            ];

            const errorIndicator = shallow(
                <ErrorIndicator clearErrors={clearErrors} errors={errors}/>);
            expect(errorIndicator).toMatchSnapshot();
        });
    });

    describe('when clicking the OK button', () => {
        it('calls the clearErrors function', () => {
            const errorIndicator = shallow(
                <ErrorIndicator clearErrors={clearErrors} errors={[]}/>);

            errorIndicator.find('ButtonInput').simulate('click');

            expect(clearErrors).toHaveBeenCalled();
        });
    });
});
