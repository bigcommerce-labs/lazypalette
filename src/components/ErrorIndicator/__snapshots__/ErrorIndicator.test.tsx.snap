// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorIndicator when rendering renders with errors 1`] = `
<styled.div>
  <p
    key="Error: hello"
  >
    Error: hello
  </p>
  <p
    key="Error: goodbye"
  >
    Error: goodbye
  </p>
  <ButtonInput
    onClick={[MockFunction]}
  >
    OK
  </ButtonInput>
</styled.div>
`;

exports[`ErrorIndicator when rendering renders with no errors 1`] = `
<styled.div>
  <ButtonInput
    onClick={[MockFunction]}
  >
    OK
  </ButtonInput>
</styled.div>
`;
